To-do 리스트 Redux를 사용하여 리팩터링 하기

kwonmory opened the pull request:
두둥.. 안보고 하려고 연습을 하다보니 늦게 제출하게 됐어요 ㅠ

kwonmory committed:
리듀서 테스트 코드 작성

kwonmory committed:
리듀서 구현

kwonmory committed:
리듀서 리펙토링

kwonmory committed:
리덕트 추가

kwonmory committed:
ListContainer 테스트 작성

kwonmory committed:
ListContainer 구현

kwonmory committed:
ListContainer 리팩토링

- 이게 리팩토링인가봉가?

kwonmory committed:
InputContainer 테스트 작성

kwonmory committed:
InputContainer 구현

kwonmory committed:
Page 컴포넌트 내용 App 으로 이동

kwonmory committed:
커버리지 100 맞추기 & 린트 적용

kwonmory committed:
Fix package-lock.json

wholemann commented on src/ListContainer.test.jsx:
```diff
@@ -0,0 +1,80 @@
+describe('ListContainer', () => {
+  context('with tasks', () => {
+    it('exist task in tasks', () => {
+      mockSelector({
```
mockSelector를 이해하려면 계속 상단을 봐야하는 부작용이 있네요.

차라리 
```javascript
useSelector.mockImplementation((selector) => selector({
    newId: 1,
    tasks,
  }));
```

는 그대로 테스트 안에 사용하고,

외부에 const tasks = [...]를 선언하고 이용하면 오히려 깔끔할 것 같네요.

useSelector.mockImplementation 자체가 해석이 어렵지 않고,

길어 보이는 건 내부에 전달하는 객체 때문인데 객체만 분리하면 될 것 같네요.

kwonmory replied:
맞아요. 이 리팩토링 한다 치고 추출해놨는데.. 나중에 다른 테스트 코드에서 taskTitle 이 추가 될 때, 재사용을 못하더라구요.. 수정하고 다시 푸쉬할께요. 아 그리고 CI 에서 통과 못하던데, 커버리지 때문인거 같아서 이것도 한번 할 수 있으면 같이 수정할께요.

wholemann reviewed (commented):
잘해주고 계십니다ㅎㅎ 역시 슬랙에서도 고민을 많이 하고 계신 걸 느낄 수 있었는데

결과물에도 잘 나오고 있어요.

한 가지 아쉬운 부분에 대해 comment 드렸으니 고민해보시고 계속 진행하셔서 과제1 언능 끝내시면 좋겠네요ㅎㅎ

kwonmory committed:
Fix reducer part

- [reducer.js] reducer 오타 수정
- [reducer.js] 데이터 추가할 때마다 input 값 비워질 수 있도록 변경
- [reducer.js] 새로운 값을 추가할 때마다 추가되는 id 가질 수 있도록 수정
- [reducer.test.js] 초기 상태가 없을 때, 테스트 코드 작성

kwonmory committed:
Fix FeedBack

- [InputContainer.test.jsx] test 구문 변경(사실 변경할 필요는 없었음..)
- [ListContainer.test.jsx] 피드백 받은 부분 수정(mockSelector 불 필요로 인해 삭제)
- [reducer.test.js] lint 수정

wholemann reviewed (approved):
잘해주셨네요.

이번에 연습하신 TDD 싸이클에 맞춰서 과제2도 진행해주세요!

고생하셨습니다.
